media-libs/svt-av1/svt-av1-0.8.4.ebuild

```diff
@@ -0,0 +1,43 @@
+# Copyright 2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit flag-o-matic cmake
+
+DESCRIPTION="Scalable Video Technology for AV1 (SVT-AV1 Encoder and Decoder)"
+HOMEPAGE="https://github.com/OpenVisualCloud/SVT-AV1"
+
+if [ ${PV} = "9999" ]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/OpenVisualCloud/SVT-AV1.git"
+else
+	SRC_URI="https://github.com/OpenVisualCloud/SVT-AV1/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64"
+	S="${WORKDIR}/SVT-AV1-${PV}"
+fi
+
+LICENSE="BSD-2"
+IUSE="debug"
+SLOT="0"
+
+DEPEND="dev-lang/nasm"
+RDEPEND=""
+
+src_prepare() {
+	append-ldflags -Wl,-z,noexecstack
+	cmake_src_prepare
+}
+
+src_configure() {
+	use debug && CMAKE_BUILD_TYPE=Debug
+	cmake_src_configure
```

Comment:
this should enable/disable unit tests with BUILD_TESTING, otherwise I think we have no tests

Reply:
Frankly dealing with tests in ebuilds is something I don't have experience with.

Enabling BUILD_TESTING requires cloning https://github.com/Cidana-Developers/aom.git/ so I think it's just best to disable them?

Reply:
isn't it possible to simply use media-libs/libaom ? it seems it clones it build the shared lib that we can pull as a dep from there

Reply:
Sorry, I never saw your reply here. That's a good idea I'll try get working. I reckon I'll close this here and try get it into GURU. Thanks

Reply:
So it looks like building the tests requires linking against a specific aom branch: https://github.com/Cidana-Developers/aom/commits/av1-normative

```
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: test/e2e_test/CMakeFiles/SvtAv1E2ETests.dir/RefDecoder.cc.o: in function `RefDecoder::~RefDecoder()':
RefDecoder.cc:(.text+0x138): undefined reference to `ifd_clear'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: test/e2e_test/CMakeFiles/SvtAv1E2ETests.dir/RefDecoder.cc.o: in function `RefDecoder::RefDecoder(RefDecoder::RefDecoderErr&, bool)':
RefDecoder.cc:(.text+0x396): undefined reference to `aom_codec_control_'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: test/e2e_test/CMakeFiles/SvtAv1E2ETests.dir/RefDecoder.cc.o: in function `RefDecoder::inspect_cb(void*, void*)':
RefDecoder.cc:(.text+0x1e03): undefined reference to `ifd_inspect'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: RefDecoder.cc:(.text+0x1e67): undefined reference to `ifd_init'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: test/e2e_test/CMakeFiles/SvtAv1E2ETests.dir/RefDecoder.cc.o: in function `RefDecoder::control(int, int)':
RefDecoder.cc:(.text+0x1ead): undefined reference to `aom_codec_control_'
```

https://github.com/Cidana-Developers/aom/commit/cfc5c9e95bcb48a5a41ca7908b44df34ea1313c0